static/app/components/dropdownMenuControlV2.tsx

```diff
@@ -121,6 +121,14 @@ function MenuControl({
     ref
   );
 
+  // Calculate the current trigger element's width. This will be used as
+  // the min width for the menu.
+  const [triggerWidth, setTriggerWidth] = useState<number>();
+  useEffect(() => {
+    const newTriggerWidth = ref.current?.offsetWidth;
+    !isSubmenu && newTriggerWidth && setTriggerWidth(newTriggerWidth);
+  }, [trigger, triggerLabel, triggerProps]);
+
```

Comment:
[Based on an existing implementation in our codebase.](https://github.com/getsentry/sentry/blob/alberto/vital-page-menu/static/app/components/charts/optionSelector.tsx#L47-L58)